main/src/library/Map.flix

```diff
@@ -126,83 +87,72 @@ namespace Map {
     /// Otherwise, updates `m` with `k -> v`.
     ///
     pub def insertWithKey(f: (k, v, v) -> v & e, k: k, v: v, m: Map[k, v]): Map[k, v] & e =
-        if (memberOf(k, m)) adjustWithKey((a, b) -> f(a, v, b), k, m)
-        else
-            let Map(xs) = m;
-                Map((k, v) :: xs)
-
-
-
-
-
+        let Map(xs) = m;
+        match get(k, m) {
+            case None => Map(RBTree.insert(k, v, xs))
+            case Some(v1) => Map(RBTree.insert(k, f(k, v, v1), xs))
+        }
 
     ///
-    /// Updates `m` with `k -> f(v)` if `k -> v` is in `m`.
+    /// Updates `m` with `k -> f(v)` if `k` is in `m`.
```

Comment:
The reason the comment was written in the old style was to make clear where `v` is coming from. Please preserve that along with other comments.